core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Array.java

```diff
@@ -0,0 +1,228 @@
+  }
+
+  public ArrayModel getArrayModel() {
+    return arrayModel;
+  }
+
+  public void setArrayModel(ArrayModel arrayModel) {
+    this.arrayModel = arrayModel;
+  }
+
+  public String getSerialNumber() {
+    return serialNumber;
+  }
+
+  public void setSerialNumber(String serialNumber) {
+    this.serialNumber = serialNumber;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public void setDescription(String description) {
+    this.description = description;
+  }
+
+  public User getCreator() {
+    return creator;
+  }
+
+  public void setCreator(User creator) {
+    this.creator = creator;
+  }
+
+  public Date getCreationTime() {
+    return creationTime;
+  }
+
+  public void setCreationTime(Date creationTime) {
+    this.creationTime = creationTime;
+  }
+
+  public User getLastModifier() {
+    return lastModifier;
+  }
+
+  public void setLastModifier(User lastModifier) {
+    this.lastModifier = lastModifier;
+  }
+
+  public Date getLastModified() {
+    return lastModified;
+  }
+
+  public void setLastModified(Date lastModified) {
+    this.lastModified = lastModified;
+  }
+
+  public Map<String, Sample> getSamples() {
+    return samples;
+  }
+
+  public void setSamples(Map<String, Sample> samples) {
+    this.samples = samples;
+  }
+
+  public Sample getSample(String position) {
+    validatePosition(position);
+    return samples.get(position);
+  }
+
+  public void setSample(String position, Sample sample) {
+    validatePosition(position);
+    if (sample == null) {
+      samples.remove(position);
+    } else {
+      samples.put(position, sample);
+    }
+  }
+
+  public boolean isPositionValid(String position) {
+    if (getArrayModel() == null) {
+      return false;
+    }
+    Matcher m = positionRegex.matcher(position);
+    if (!m.matches()) {
+      return false;
+    }
+    int y = Integer.parseInt(m.group(1));
+    int x = Integer.parseInt(m.group(2));
+    return isInBounds(x, y);
+  }
+
+  private void validatePosition(String position) {
+    if (getArrayModel() == null) {
+      throw new IllegalStateException("ArrayModel not set. Cannot verify size of Array.");
+    }
+    Matcher m = positionRegex.matcher(position);
+    if (!m.matches()) {
+      throw new IllegalArgumentException("Invalid Array position String: " + position);
+    }
+    int y = Integer.parseInt(m.group(1));
+    int x = Integer.parseInt(m.group(2));
+    if (!isInBounds(x, y)) {
+      throw new IndexOutOfBoundsException("Position " + position + " is outside of the Array bounds");
+    }
+  }
+
+  private boolean isInBounds(int x, int y) {
+    return x > 0 && x <= getArrayModel().getColumns() && y > 0 && y <= getArrayModel().getRows();
+  }
+
+  @Override
+  public Collection<ChangeLog> getChangeLog() {
+    return changeLog;
+  }
+
+  @Override
+  public ChangeLog createChangeLog(String summary, String columnsChanged, User user) {
+    ArrayChangeLog changeLog = new ArrayChangeLog();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "changeLog" which hides the field declared at line 82. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)